core/src/client/mod.rs

```diff
@@ -675,4 +698,516 @@ impl StarknetClient for StarknetClientImpl {
+                            // PendingBlockWithTxs doesn't have a block hash
+                            hash: None,
+                            parent_hash,
+                            uncles_hash: parent_hash,
+                            author: sequencer,
+                            miner: sequencer,
+                            // PendingBlockWithTxs doesn't have a state root
+                            state_root: PrimitiveH256::zero(),
+                            // PendingBlockWithTxs doesn't have a transactions root
+                            transactions_root: PrimitiveH256::zero(),
+                            // PendingBlockWithTxs doesn't have a receipts root
+                            receipts_root: PrimitiveH256::zero(),
+                            // PendingBlockWithTxs doesn't have a block number
+                            number: None,
+                            gas_used,
+                            gas_limit,
+                            extra_data,
+                            logs_bloom,
+                            timestamp,
+                            difficulty,
+                            nonce,
+                            size,
+                            base_fee_per_gas,
+                            mix_hash,
+                        };
+                        let block = Block {
+                            header,
+                            total_difficulty,
+                            uncles: vec![],
+                            transactions,
+                            base_fee_per_gas: None,
+                            size,
+                        };
+                        Ok(Rich::<Block> {
+                            inner: block,
+                            extra_info: BTreeMap::default(),
+                        })
+                    }
+                    MaybePendingBlockWithTxs::Block(block_with_txs) => {
+                        println!("1. Calling Block With Txs");
+                        let hash =
+                            PrimitiveH256::from_slice(&block_with_txs.block_hash.to_bytes_be());
+                        let parent_hash =
+                            PrimitiveH256::from_slice(&block_with_txs.parent_hash.to_bytes_be());
+                        let sequencer = H160::from_slice(
+                            &block_with_txs.sequencer_address.to_bytes_be()[12..32],
+                        );
+                        let state_root =
+                            PrimitiveH256::from_slice(&block_with_txs.new_root.to_bytes_be());
+                        let transactions_root = PrimitiveH256::from_slice(
+                            &"0xac91334ba861cb94cba2b1fd63df7e87c15ca73666201abd10b5462255a5c642"
+                                .as_bytes()[1..33],
+                        );
+                        let receipts_root = PrimitiveH256::from_slice(
+                            &"0xf2c8755adf35e78ffa84999e48aba628e775bb7be3c70209738d736b67a9b549"
```

Comment:
Why do we need this?

Reply:
Same as transactions_root

Reply:
Let's try to replace it by PrimitiveH256::zero for now

Reply:
Ok